source/patterns/@aws-solutions-constructs/core/lib/elasticsearch-helper.ts

```diff
@@ -217,3 +217,115 @@ export function buildElasticSearchCWAlarms(scope: Construct): cloudwatch.Alarm[]
 
   return alarms;
 }
+
+function retrievePrivateSubnetIds(vpc: ec2.IVpc) {
+  let targetSubnetType;
+
+  if (vpc.isolatedSubnets.length) {
+    targetSubnetType = ec2.SubnetType.PRIVATE_ISOLATED;
+  } else if (vpc.privateSubnets.length) {
+    targetSubnetType = ec2.SubnetType.PRIVATE_WITH_NAT;
+  } else {
+    throw new Error('Error - ElasticSearch Domains can only be deployed in Isolated or Private subnets');
+  }
+
+  const subnetSelector = {
+    onePerAz: true,
+    subnetType: targetSubnetType
+  };
+
+  return vpc.selectSubnets(subnetSelector).subnetIds;
+}
+
+function createClusterConfiguration(zoneLength?: number): elasticsearch.CfnDomainProps {
+  let clusterConfig = {};
+
+  if (zoneLength === 0) { // Public, Zone Awareness is Enabled
+    clusterConfig = {
+      dedicatedMasterEnabled: true,
+      dedicatedMasterCount: 3,
+      zoneAwarenessEnabled: true,
+      zoneAwarenessConfig: {
+        availabilityZoneCount: 3
+      },
+      instanceCount: 3,
+    };
+  } else if (zoneLength === 1) { // VPC, Zone Awareness is Disabled
```

Comment:
Never a possibility - we will never create a single zone domain unless the client sends us a cluster config. We'll never create a cluster config for a single zone domain.